### DISCUSSION TITLE
Support ESLint Flat Config

### DISCUSSION THREAD

**Discussion by shinGangan** [chakra-ui/panda#1686] [Ideas]
### Description

In ESLint v8.0.0, only eslintrc format is supported.
the new flat format ‘Flat Config’ will be the default in ESLint v9.0.0
- https://eslint.org/blog/2023/10/flat-config-rollout-plans/

### Problem Statement/Justification

Hi @segunadebayo and maintainer team.
Thanks to the great work of the maintainer team 👏

In ESLint v8.0.0, only eslintrc format is supported.
the new flat format ‘Flat Config’ will be the default in ESLint v9.0.0.

- https://eslint.org/blog/2023/10/flat-config-rollout-plans/

Do you have plans for supporting flat config in Panda CSS?
I would be happy to help if you plan to support to do.

### Proposed Solution or API

We are experimenting with the following on [this branch](https://github.com/shinGangan/panda/tree/exp/support-eslint-flat-config)

- Support ESLint Flat Config
  - add `eslint.config.js`
  - delete `.eslintrc.js`, `.eslintignore`
- Migrate @typescript-eslint from v5 to v6: https://typescript-eslint.io/blog/announcing-typescript-eslint-v6/
  - Enable `recommended`, `stylistic` ('typescript-eslint' team recommended).
  - Enable only rules that are the same as the v5 recommended rules.
  - Turn off rules added by `stylistic` once so that they do not affect the roadmap

If there are no problems, I would like to submit a Pull Request.
`/playground` and `/sandbox` would like to supported by another Pull Requests

## ESLint

- https://eslint.org/blog/2023/10/flat-config-rollout-plans/
- https://github.com/vuejs/eslint-plugin-vue/pull/2226
- https://github.com/typescript-eslint/typescript-eslint/pull/6836

## @typescript-eslint

- https://typescript-eslint.io/blog/announcing-typescript-eslint-v6/
- https://github.com/typescript-eslint/typescript-eslint/discussions/6014

**Comment by segunadebayo** [comment]
Hey @shinGangan,

We're definitely open to this. Feel free to open a PR

**Comment by shinGangan** [comment]
@segunadebayo , thank you.
Glad to be of help the Chakra UI team.

Supporting Flat Config and migrating 'typescript-eslint' to v6 are different in content. So I will submit a separate Pull Request.
